pgx/src/hooks.rs

```diff
@@ -458,6 +485,61 @@ unsafe extern "C" fn pgx_planner_impl(
     .inner
 }
 
+#[cfg(any(feature = "pg10", feature = "pg11", feature = "pg12", feature = "pg13"))]
+#[pg_guard]
+unsafe extern "C" fn pgx_post_parse_analyze(
+    parse_state: *mut pg_sys::ParseState,
+    query: *mut pg_sys::Query,
+) {
+    fn prev(
+        parse_state: PgBox<pg_sys::ParseState>,
+        query: PgBox<pg_sys::Query>,
+        _jumble_state: Option<PgBox<JumbleState>>,
+    ) -> HookResult<()> {
+        HookResult::new(unsafe {
+            match HOOKS.as_mut().unwrap().prev_post_parse_analyze_hook.as_ref() {
+                None => (),
+                Some(f) => (f)(parse_state.as_ptr(), query.as_ptr()),
+            }
+        })
+    }
+
+    let hook = &mut HOOKS.as_mut().unwrap().current_hook;
+    hook.post_parse_analyze(PgBox::from_pg(parse_state), PgBox::from_pg(query), None, prev).inner
+}
+
+#[cfg(any(feature = "pg14"))]
+#[pg_guard]
+unsafe extern "C" fn pgx_post_parse_analyze(
+    parse_state: *mut pg_sys::ParseState,
+    query: *mut pg_sys::Query,
+    jumble_state: *mut JumbleState,
+) {
+    fn prev(
+        parse_state: PgBox<pg_sys::ParseState>,
+        query: PgBox<pg_sys::Query>,
+        jumble_state: Option<PgBox<JumbleState>>,
+    ) -> HookResult<()> {
+        HookResult::new(unsafe {
+            match HOOKS.as_mut().unwrap().prev_post_parse_analyze_hook.as_ref() {
+                None => (),
+                Some(f) => {
+                    (f)(parse_state.as_ptr(), query.as_ptr(), jumble_state.unwrap().as_ptr())
+                }
+            }
+        })
+    }
+
+    let hook = &mut HOOKS.as_mut().unwrap().current_hook;
+    hook.post_parse_analyze(
+        PgBox::from_pg(parse_state),
+        PgBox::from_pg(query),
+        Some(PgBox::from_pg(jumble_state)),
+        prev,
+    )
+    .inner
+}
+
```

Comment:
This would eliminate the duplication, and works even with #794 not being fixed, but considering the macro magic, I am not sure it's worth it. What do you think?

```suggestion
macro_rules! pgx_post_parse_analyze_body {
    ($parse_state: ident, $query: ident$(, $_jumble_state: ident)?) => {
        {
            fn prev(
                parse_state: PgBox<pg_sys::ParseState>,
                query: PgBox<pg_sys::Query>,
                _jumble_state: Option<PgBox<JumbleState>>,
            ) -> HookResult<()> {
                HookResult::new(unsafe {
                    match HOOKS.as_mut().unwrap().prev_post_parse_analyze_hook.as_ref() {
                        None => (),
                        #[cfg(any(feature = "pg10", feature = "pg11", feature = "pg12", feature = "pg13"))]
                        Some(f) => (f)(parse_state.as_ptr(), query.as_ptr()),
                        #[cfg(any(feature = "pg14"))]
                        Some(f) => (f)(parse_state.as_ptr(), query.as_ptr(), _jumble_state.unwrap().as_ptr()),
                    }
                })
            }
            #[cfg(any(feature = "pg10", feature = "pg11", feature = "pg12", feature = "pg13"))]
            let jumble_box = None;
            #[cfg(any(feature = "pg14"))]
            let jumble_box =  Some(PgBox::from_pg($($_jumble_state)?));

            let hook = &mut HOOKS.as_mut().unwrap().current_hook;
            hook.post_parse_analyze(
                PgBox::from_pg($parse_state),
                PgBox::from_pg($query),
                jumble_box,
                prev,
            )
            .inner
        }
    }
}

#[cfg(any(feature = "pg10", feature = "pg11", feature = "pg12", feature = "pg13"))]
#[pg_guard]
unsafe extern "C" fn pgx_post_parse_analyze(
    parse_state: *mut pg_sys::ParseState,
    query: *mut pg_sys::Query,
) {
    pgx_post_parse_analyze_body!(parse_state, query)
}

#[cfg(any(feature = "pg14"))]
#[pg_guard]
unsafe extern "C" fn pgx_post_parse_analyze(
    parse_state: *mut pg_sys::ParseState,
    query: *mut pg_sys::Query,
    _jumble_state: *mut JumbleState,
) {
    pgx_post_parse_analyze_body!(parse_state, query, _jumble_state)
}

```

Reply:
Jubilee and I discussed this on discord and agreed that this is probably more magic than it's worth.